src/test/java/com/xebia/headerbuddy/e2e/HeaderBuddyApiKeyTest.java

```diff
@@ -0,0 +1,62 @@
+package com.xebia.headerbuddy.e2e;
+
+import com.xebia.headerbuddy.models.entities.Euser;
+import com.xebia.headerbuddy.models.entities.repositories.EuserRepository;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.Date;
+
+// for the annotations
+@RunWith(SpringRunner.class)
+// Run spring on a random port
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class HeaderBuddyApiKeyTest {
+
+    @Autowired
+    private TestRestTemplate template;
+    @Autowired
+    private EuserRepository userRepository;
+
+    // Get the random port spring is running on
+    @Value("${local.server.port}")
```

Comment:
See comment below this is not necessary if you change the call with the `TestRestTemplate`